app/views/katello/api/v2/content_view_repositories/show_all.json.rabl

```diff
@@ -0,0 +1,8 @@
+object false
+
+extends "katello/api/v2/common/metadata"
+
+child @collection[:results] => :results do
+  extends 'katello/api/v2/repositories/base'
+  node(:added_to_content_view) { |repo| repo.in_content_view?(@content_view) }
```

Comment:
Won't this add a SQL query for each result row?  Perhaps the query in `content_view_repositories_controller#show_all` could be even further improved by using Active Record [`.select`](https://guides.rubyonrails.org/active_record_querying.html#selecting-specific-fields), something like

```rb
query = Katello::Repository.readable.in_default_view.in_organization(@organization).select("#{kr}.*, #{krr}.*, #{kcvr}.*, (#{kcvr}.repository_id IS NOT NULL) AS added")
```

(not sure if that condition is correct, double-check me)

Then each result will gain an `.added` method that returns true or false.  So here, in your rabl, you could just say 
```rb
attributes :added
```
with no need to call a method that runs more SQL.  Hopefully this would still work with `scoped_search`..  :crossed_fingers: 

And as a side bonus, it would simplify the `order_query` so you could just say `.order("added")`

Reply:
Because @content_view is an instance variable, its repositories association should get lazy only once, and not cause a query per repo, unless i misreading something.  Does that make sense?

Reply:
@jlsherrill aah yeah, I didn't think about that! Makes sense.

Reply:
@jeremylenz are you all set with this or would like some changes here? 

The issue I ran into with using a custom column is we ultimately put everything in a subquery since we [re-query by ID](https://github.com/Katello/katello/blob/c50717c04f6b885d91da4280ae5b0de59cddc5b9/app/controllers/katello/api/v2/api_controller.rb#L81) so a custom column like that is not accessible when it comes to the rabl. I tried to modify scoped_search, but it didn't prove to be easy without making breaking changes.

This was the source of a lot of trouble here, but doing everything in the custom_search including the ordering allowed it to work as expected and be sorted. It does require the method to be called in the rabl to tell if it was added_for_repo, but I don't see a huge performance hit with that, we actually do a lot of queries per-repo, here are some for just one repo, notice they take 1ms or less: (not that we couldn't speed up this and many other db queries in our app :smile:)

```
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/models/katello/repository.rb:721:in `docker_meta_tag_count'
2020-08-20T00:56:34 [D|sql|31374920]    (0.8ms)  SELECT COUNT(*) FROM "katello_rpms" INNER JOIN "katello_repository_rpms" ON "katello_rpms"."id" = "katello_repository_rpms"."rp
m_id" WHERE "katello_repository_rpms"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:33:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (1.0ms)  SELECT COUNT(*) FROM "katello_srpms" INNER JOIN "katello_repository_srpms" ON "katello_srpms"."id" = "katello_repository_srpms"
."srpm_id" WHERE "katello_repository_srpms"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:34:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]   CACHE  (0.0ms)  SELECT COUNT(*) FROM "katello_rpms" INNER JOIN "katello_repository_rpms" ON "katello_rpms"."id" = "katello_repository_rpm
s"."rpm_id" WHERE "katello_repository_rpms"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:35:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.7ms)  SELECT COUNT(*) FROM "katello_package_groups" INNER JOIN "katello_repository_package_groups" ON "katello_package_groups"."id" =
 "katello_repository_package_groups"."package_group_id" WHERE "katello_repository_package_groups"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:36:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.7ms)  SELECT COUNT(*) FROM "katello_errata" INNER JOIN "katello_repository_errata" ON "katello_errata"."id" = "katello_repository_err
ata"."erratum_id" WHERE "katello_repository_errata"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:37:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.7ms)  SELECT COUNT(*) FROM "katello_puppet_modules" INNER JOIN "katello_repository_puppet_modules" ON "katello_puppet_modules"."id" =
 "katello_repository_puppet_modules"."puppet_module_id" WHERE "katello_repository_puppet_modules"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:38:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.9ms)  SELECT COUNT(*) FROM "katello_files" INNER JOIN "katello_repository_file_units" ON "katello_files"."id" = "katello_repository_f
ile_units"."file_unit_id" WHERE "katello_repository_file_units"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:39:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.6ms)  SELECT COUNT(*) FROM "katello_debs" INNER JOIN "katello_repository_debs" ON "katello_debs"."id" = "katello_repository_debs"."de
b_id" WHERE "katello_repository_debs"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:40:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.7ms)  SELECT COUNT(*) FROM "katello_module_streams" INNER JOIN "katello_repository_module_streams" ON "katello_module_streams"."id" =
 "katello_repository_module_streams"."module_stream_id" WHERE "katello_repository_module_streams"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:41:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]    (0.7ms)  SELECT COUNT(*) FROM "katello_ansible_collections" INNER JOIN "katello_repository_ansible_collections" ON "katello_ansible_collections"."id" = "katello_repository_ansible_collections"."ansible_collection_id" WHERE "katello_repository_ansible_collections"."repository_id" = $1  [["repository_id", 110]]
2020-08-20T00:56:34 [D|sql|31374920]   ↳ /home/vagrant/katello/app/views/katello/api/v2/repositories/base.json.rabl:42:in `block in cached_source_450191144403171490'
2020-08-20T00:56:34 [D|sql|31374920]   Katello::Repository Exists? (1.1ms)  SELECT 1 AS one FROM "katello_repositories" INNER JOIN "katello_content_view_repositories" ON "katello_repositories"."id" = "katello_content_view_repositories"."repository_id" WHERE "katello_content_view_repositories"."content_view_id" = $1 AND "katello_repositories"."id" = $2 LIMIT $3  [["content_view_id", 10], ["id", 110], ["LIMIT", 1]]
```

Reply:
After Justin's explanation I think my concern is addressed.. I'm good with this :+1: